src/components/Common/Filters/QuickFilter.vue

```diff
@@ -109,35 +109,37 @@ export default {
     submitOnType: {
       type: Boolean,
       default: true
+    },
+    initialValue: {
+      type: String,
+      default: ''
     }
   },
   data() {
     return {
-      inputSearchTerm: this.searchTerm
+      value: this.initialValue
     }
   },
   methods: {
     submitSearch() {
-      this.$emit('update-filter', this.inputSearchTerm)
+      this.$emit('update-filter', this.value)
     },
     resetSearch() {
       this.$emit('reset')
     },
     displayAdvancedFilters() {
       this.$emit('display-advanced-filters')
+    },
+    inputSubmit () {
+      this.$emit('enter-pressed')
     }
   },
   watch: {
-    inputSearchTerm() {
-      if (!this.submitOnType) {
-        return
-      }
-      this.submitSearch()
-    },
-    searchTerm: {
-      immediate: true,
-      handler(val) {
-        this.inputSearchTerm = val
+    value () {
+      this.$emit('input', this.value)
```

Comment:
And YES! We should maximize the use of the `input` event and the `value` prop, so that components using our custom components can leverage the `v-model` macro-attribute